File: plugin/src/integrationTest/resources/project/lib/src/test/java/org/test/lib/LibOwnersTest.java
package org.test.lib;

import org.junit.Test;
import org.test.utils.LibUtils;

import static io.github.gmazzo.codeowners.CodeOwners.getCodeOwners;

import static kotlin.collections.SetsKt.setOf;
import static org.junit.Assert.assertEquals;

public class LibOwnersTest {

    @Test
    public void ownerOfLib() {
        assertEquals(setOf("kotlin-devs"), getCodeOwners(LibClass.class));
    }

    @Test
    public void ownerOfUtils() {
        assertEquals(setOf("kotlin-devs"), getCodeOwners(LibUtils.class));
    }

}

File: demo-project-jvm/lib1/src/test/kotlin/org/test/jvm/lib/LibOwnersTest.kt
package org.test.jvm.lib

import io.github.gmazzo.codeowners.codeOwners
import kotlin.test.Test
import kotlin.test.assertEquals
import org.test.jvm.utils.LibUtils
import org.test.jvm.utils.more.MoreUtils

class LibOwnersTest {

    @Test
    fun ownerOfSelf() {
        assertEquals(setOf("jvm-libs-devs"), LibOwnersTest::class.java.codeOwners)
    }

    @Test
    fun ownerOfLib() {
        assertEquals(setOf("jvm-libs-devs"), LibClass::class.java.codeOwners)
    }

    @Test
    fun ownerOfUtils() {
        assertEquals(setOf("jvm-libs-devs"), LibUtils::class.java.codeOwners)
    }

    @Test
    fun ownerOfMoreUtils() {
        assertEquals(setOf("jvm-utils-devs"), MoreUtils::class.java.codeOwners)
        assertEquals(setOf("jvm-utils-devs"), MoreUtils.Companion::class.java.codeOwners)
    }

}
